### PR TITLE
RHMAP-9235 - Moving from wildfly to EAP

### PR DESCRIPTION
**Pull request by wtrocki**
# Motivation

Use EAP image as base instead of wildfly. Updated image to use most recent EAP. 
Image requires RHEL/subscription for tests.

ping @matzew

### CODE REVIEW

**Comment by wtrocki** [comment]
Images were pushed to the dockerhub now. See jenkins job:
http://bob.feedhenry.net:8080/job/unifiedpush-server_triggered_image_build/34/console

Going to verify image locally and create templates PR for verification purposes.

**Comment by wtrocki** [comment]
@matzew Can you look into this changes?
It looks like even eap6 docker images are running now on java8. 
Server is running but both ups and keyloak are not loaded.
How difficult would be to merge master into current product in order to get latest java and run this on eap7?

**Comment by matzew** [comment]
@wtrocki hrm, I am surprised there is no EAP6/OpenJDK7 image. Red Hat does have support for a few more years for OpenJDK.

For only on-prem, we could do a special branch, of fheng/a-up-s

But I guess an update to Java-8 and 1.2.0.x UPS is a no-go...

In theory we could introduce a second JVM (Java-8) only for EAP there, since Millicore does not run on java8 - it will be required when we decouple our internal Keycloak build - we can speak in the meeting ;-)

**Comment by wtrocki** [comment]
@matzew Thanks for response. Millicore was migrated to run on java8 (Compiled to java7 runtime level) so we can (and should) update our clusters to java8 as well.

**Comment by matzew** [comment]
A few initial thoughts

* the `system-properties` is lacking the `keycloak.host` value (from SaaS config)

in the config,
```
         <socket-binding name="ajp" port="${jboss.ajp.port:8009}"/>
        <socket-binding name="http" port="${jboss.http.port:8080}"/>
        <socket-binding name="https" port="${jboss.https.port:8443}"/>
```

locally the values are hard-coded, not as variables

File: eap/unifiedpush-eap/Dockerfile [hunk 1]
```diff
@@ -24,11 +24,11 @@ RUN mv aerogear-unifiedpush-server*/* .  && \
...
 
 # Switch to the working dir $JBOSS_HOME/standalone/deployments
-WORKDIR /opt/jboss/wildfly/standalone/deployments
+WORKDIR $JBOSS_HOME/standalone/deployments
 
 # copy war files
-RUN cp $UPSDIST/servers/unifiedpush-auth-server.war $JBOSS_HOME/standalone/deployments \
-    && cp $UPSDIST/servers/unifiedpush-server-wildfly.war $JBOSS_HOME/standalone/deployments
+RUN cp $UPSDIST/servers/unifiedpush-auth-server.war . \
+    && cp $UPSDIST/servers/unifiedpush-server-wildfly.war .
```
Comment (matzew) [inline eap/unifiedpush-eap/Dockerfile]: It should be `unifiedpush-server-as7.war`, instead of -wildfly for the UPS WAR file @wtrocki

**Comment by wtrocki** [comment]
@matzew I build image and it looks like everything works. 
I deployed changes to our testing cluster so you can verify changes.

File: eap/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
 # Run everything below as root user
 USER root
+ 
 
-# Clean the metadata
-RUN yum install -y unzip wget && yum -q clean all
+# Install required packages. We need to enable only required repo
+RUN yum install --disablerepo * --enablerepo rhel-7-server-rpms  -y unzip wget  && \
+	yum -q clean all
```
Comment (matzew) [inline eap/Dockerfile]: formatting

Comment (matzew) [inline eap/Dockerfile]: not sure we want TABs :)

**Comment by matzew** [comment]
👍  works perfect

**Comment by wtrocki** [comment]
@matzew Thansk. Going to fix style and unify image in separate PR.

**Pull request merged**